controller/Api/EventBus/IEvent.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Text;
+
+namespace SynthesisAPI.EventBus
+{
+    public interface IEvent 
+    {
+        public string EventType { get; }
+
+        public byte[] getData();
```

Comment:
I would get rid of this. Any data that would be passed in a IEvent type would just be in the form of properties that would be declared inside the newly defined IEvent type.

Reply:
Check my new changes I've pushed. I added general parametric support for arguments.